src/main/java/com/diffplug/gradle/eclipse/apt/AptEclipsePlugin.java

```diff
@@ -103,14 +119,13 @@ private void configureJdtApt(
                     .getExtensions()
                     .getByType(AptPlugin.AptOptions.class)
                     .isAnnotationProcessing()));
-    jdtApt.setProcessorOptions(
-        () ->
-            project
-                .getTasks()
-                .getByName(mainSourceSet.getCompileJavaTaskName())
-                .getExtensions()
-                .getByType(AptPlugin.AptOptions.class)
-                .getProcessorArgs());
+    MapProperty<String, ?> processorOptions = objects
+      .mapProperty(String.class, Object.class)
+      .value(project.getTasks().named(mainSourceSet.getCompileJavaTaskName()).map(task ->
+        task.getExtensions().getByType(AptPlugin.AptOptions.class).getProcessorArgs()
+      ));
+    jdtApt.setProcessorOptions(processorOptions::getOrNull);
+    project.afterEvaluate(p -> processorOptions.finalizeValue());
```

Comment:
Using a `MapProperty` object here allows Gradle to finalize its value before task execution. In configuration cache mode, all references to `Project` in objects are nullified, so this would cause an NPE unless configuration cache was disabled.